### PR TITLE
nspr: compile support for apple silicon

### PR DESCRIPTION
**Pull request by timblechmann**
Specify library name and version:  **nspr/4.27**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

---

i'm not sure how to deal with the test package correctly: on apple silicon the test package is still compiled as `x86_64` not as `arm64`, so linking fails

### CODE REVIEW

**Comment by timblechmann** [comment]
note: contains the version bump from #2305

File: recipes/nspr/all/patches/0001-apple-silicon-support.patch [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
+   --libdir=DIR            object code libraries [EPREFIX/lib]
+   --includedir=DIR        C header files [PREFIX/include]
+   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
+@@ -6448,7 +6436,7 @@ fi
+     CFLAGS="$CFLAGS -Wall -fno-common"
+     case "${target_cpu}" in
+         arm*)
+-            CPU_ARCH=arm
++            CPU_ARCH=arm64
```
Comment (SSE4) [inline recipes/nspr/all/patches/0001-apple-silicon-support.patch:69]: will not it affect (break) regular 32-bit arm compication (armv7, armv7s, armv7k, etc)?

Comment (timblechmann) [reply recipes/nspr/all/patches/0001-apple-silicon-support.patch:69]: this codepath is darwin-only. afaict the last iOS device supporting armv7 was the iPhone 5 / A6 processor released in 2012

Comment (SSE4) [reply recipes/nspr/all/patches/0001-apple-silicon-support.patch:69]: even with darwin, I guess watchOS/tvOS still may use it, iOS is not only the one

Comment (madebr) [reply recipes/nspr/all/patches/0001-apple-silicon-support.patch:69]: Weird how `configure.in` only uses `AC_CANONICAL_TARGET` and not `AC_CANONICAL_HOST`.

Would this work?
```diff
+         arm64)
+            CPU_ARCH=arm64
+            ;;
          arm*)
             CPU_ARCH=arm
              ;;
          i*86*|x86_64)
```

Comment (madebr) [reply recipes/nspr/all/patches/0001-apple-silicon-support.patch:69]: @timblechmann 
Have you tried my suggestion?

Comment (danimtb) [reply recipes/nspr/all/patches/0001-apple-silicon-support.patch:69]: ping @timblechmann, please give it a go and let's try to move this PR forward. Thanks!

**Comment by timblechmann** [comment]
declining in favour of #3795

**Pull request closed without merging**